src/components/CardErrorBoundary/card-error-boundary.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+// adapted from https://mui.com/material-ui/react-card/#complex-interaction
+
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Card from '@mui/material/Card';
+import CardHeader from '@mui/material/CardHeader';
+import CardContent from '@mui/material/CardContent';
+import CardActions from '@mui/material/CardActions';
+import Collapse from '@mui/material/Collapse';
+import IconButton from '@mui/material/IconButton';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import ReplayIcon from '@mui/icons-material/Replay';
+import Typography from '@mui/material/Typography';
+import { styled } from '@mui/material/styles';
+import { FormattedMessage } from 'react-intl';
+
+const ExpandMore = styled((props) => {
+    const { expand, ...other } = props;
+    return <IconButton {...other} />;
+})(({ theme, expand }) => ({
+    transform: !expand ? 'rotate(0deg)' : 'rotate(180deg)',
+    marginLeft: 'auto',
+    transition: theme.transitions.create('transform', {
+        duration: theme.transitions.duration.shortest,
+    }),
+}));
+
+class CardErrorBoundary extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { hasError: false };
+        this.handleExpandClick = this.handleExpandClick.bind(this);
+        this.handleReloadClick = this.handleReloadClick.bind(this);
+    }
+
+    static getDerivedStateFromError(error) {
+        // Update state so the next render will show the fallback UI.
+        return { hasError: true, error };
+    }
+
+    componentDidCatch(error, errorInfo) {
+        // You can also log the error to an error reporting service
+        console.error('CardErrorBoundary caught: ', error, errorInfo);
+    }
+
+    handleExpandClick() {
+        this.setState((state) => ({ ...state, expanded: !state.expanded }));
+    }
+
+    handleReloadClick() {
+        this.setState((state) => ({ hasError: false }));
```

Comment:
reloading is only setting the error to false ? No page refresh ? I'm not sure about the app behaviour in case of a breaking error

Reply:
everything has been unmoutned when there is an errror, so when setting to false it remounts everything. States inside components are reset if you want (but not redux or states higher up the tree)